src/Aspire.Hosting/DistributedApplication.cs

```diff
@@ -32,13 +64,75 @@ public DistributedApplication(IHost host)
+    /// passed to the <see cref="CreateBuilder()"/> method the AppHost has no
+    /// way to be put into publish mode. Refer to <see cref="CreateBuilder(string[])"/> or  <see cref="CreateBuilder(DistributedApplicationOptions)"/>
+    /// when more control is needed over the behavior of the distributed application at runtime.
+    /// </remarks>
+    /// <example>
+    /// The following example shows creating a Postgres server resource with a database and referencing that
+    /// database in a .NET project.
+    /// <code>
+    /// var builder = DistributedApplication.CreateBuilder();
+    /// var inventoryDatabase = builder.AddPostgres("mypostgres").AddDatabase("inventory");
+    /// builder.AddProject&lt;Projects.InventoryService&gt;()
+    ///        .WithReference(inventoryDatabase);
+    ///
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
     public static IDistributedApplicationBuilder CreateBuilder() => CreateBuilder([]);
 
     /// <summary>
     /// Creates a new instance of <see cref="IDistributedApplicationBuilder"/> with the specified command-line arguments.
     /// </summary>
     /// <param name="args">The command-line arguments to use when building the distributed application.</param>
     /// <returns>A new instance of <see cref="IDistributedApplicationBuilder"/>.</returns>
+    /// <remarks>
+    /// <para>
+    /// The <see cref="DistributedApplication.CreateBuilder(string[])"/> method is the most common way to
+    /// create an instance of the <see cref="IDistributedApplicationBuilder"/> interface. Typically this
+    /// method will be called as a top-level statement in the application's entry-point.
+    /// </para>
+    /// <para>
+    /// Note that the <paramref name="args"/> parameter is a string is essential in allowing the application
+    /// host to work with deployment tools because arguments are used to tell the application host that it
+    /// is in publish mode. If <paramref name="args"/> is not provided the application will not work with
+    /// deployment tools. It is also possible to provide arguments using the <see cref="CreateBuilder(Aspire.Hosting.DistributedApplicationOptions)"/>
+    /// overload of this method.
+    /// </para>
+    /// <example>
+    /// The following example shows creating a Postgres server resource with a database and referencing that
+    /// database in a .NET project.
+    /// <code>
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    /// var inventoryDatabase = builder.AddPostgres("mypostgres").AddDatabase("inventory");
+    /// builder.AddProject&lt;Projects.InventoryService&gt;()
+    ///        .WithReference(inventoryDatabase);
+    ///
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    /// <example>
+    /// The following example is equivalent to the previous example except that it does not use top-level statements.
+    /// <code>
+    /// public class Program
```

Comment:
Why not use a top-level program here, instead of Program/Main?

Reply:
Intentional. There is two examples here, one is top level the other isn't The key point I'm trying to make here is you need to plumb through the args (otherwise publishing won't work).